inc/template-tags.php

```diff
@@ -100,3 +100,36 @@ function _s_entry_footer() {
 	);
 }
 endif;
+
+if ( ! function_exists( '_s_post_thumbnail' ) ) :
+/**
+ * Displays an optional post thumbnail.
+ *
+ * Wraps the post thumbnail in an anchor element on index views, or a div
+ * element when on single views.
+ */
+function _s_post_thumbnail() {
+	if ( post_password_required() || is_attachment() || ! has_post_thumbnail() ) {
+		return;
+	}
+
+	if ( is_singular() ) :
+	?>
+
+	<div class="post-thumbnail">
+		<?php the_post_thumbnail(); ?>
+	</div><!-- .post-thumbnail -->
+
+	<?php else : ?>
+
+	<a class="post-thumbnail" href="<?php the_permalink(); ?>" aria-hidden="true">
+		<?php
+			the_post_thumbnail( 'post-thumbnail', array(
+				'alt' => the_title_attribute( 'echo=0' )
```

Comment:
@grappler It's correct as coded. It can be confusing though because of the way the image is used. In this case it's linked, which means that the alt text should be the purpose of the link – in this case, the post title. See: http://webaim.org/techniques/alttext/